pyccel/parser/semantic.py

```diff
@@ -944,7 +946,8 @@ def _assign_lhs_variable(self, lhs, d_var, rhs, new_expressions, is_augassign, *
 
             d_lhs = d_var.copy()
             # ISSUES #177: lhs must be a pointer when rhs is heap array
-            self._ensure_target(rhs, d_lhs)
+            if not arr_in_multirets:
+                self._ensure_target(rhs, d_lhs)
```

Comment:
This `if` also needs adding round the other 2 places where this function is called